nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from typing import Dict, List, Optional, Union
+
+from omegaconf import DictConfig, ListConfig, OmegaConf, open_dict
+from pytorch_lightning import Trainer
+
+from nemo.collections.asr.metrics.wer_bpe import WERBPE, CTCBPEDecoding, CTCBPEDecodingConfig
+from nemo.collections.asr.models.rnnt_models import EncDecRNNTBPEModel
+from nemo.utils import logging, model_utils
+
+
+class EncDecHybridRNNTCTCBPEModel(EncDecRNNTBPEModel):
+    """Base class for encoder decoder RNNT-based models with auxiliary CTC decoder/loss and subword tokenization."""
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        # Convert to Hydra 1.0 compatible DictConfig
+        super().__init__(cfg=cfg, trainer=trainer)
+        cfg = model_utils.convert_model_config_to_dict_config(cfg)
+        cfg = model_utils.maybe_update_config_version(cfg)
+
+        # setup auxiliary CTC decoder if needed
+        if 'aux_ctc' in cfg:
```

Comment:
No need for if, assert that its there in config.

Reply:
I can drop it but being here makes it a class which can also being used to just train an RNNT. I know that we have separate class, but kept is here to have more general model. I can drop it if not needed.

Reply:
raised an error when aux_ctc is not found.